### PR TITLE
Add support for Bitfinity Mainnet

### PR DESCRIPTION
**Pull request by octo-gone**

### CODE REVIEW

File: sqa/eth/ingest/ingest.py [hunk 1]
```diff
@@ -364,6 +379,17 @@ async def _fetch_single_tx_receipt(self, tx: Transaction) -> None:
 
         if self._is_skale_nebula:
             receipt['type'] = '0x0'
+        if self._is_bitfinity_mainnet:
+            if receipt.get('type') is None:
+                tx: Receipt = await self._rpc.call(
```
Comment (tmcgroul) [inline sqa/eth/ingest/ingest.py:384]: wrong type annotation

**Review by tmcgroul — CHANGES_REQUESTED** [review]

File: sqa/eth/ingest/ingest.py [hunk 2]
```diff
@@ -216,6 +218,14 @@ async def _fetch_blocks(self, from_block: int, to_block: int) -> list[Block]:
             for block in blocks:
                 for tx in block['transactions']:
                     tx['type'] = '0x0'
+        if self._is_bitfinity_mainnet:
+            for block in blocks:
+                for tx in block['transactions']:
+                    if tx.get('type') is None:
+                        if 'maxPriorityFeePerGas' in tx:
+                            tx['type'] = '0x2'
```
Comment (tmcgroul) [inline sqa/eth/ingest/ingest.py:226]: i have an idea that this assuption about tx type shoul be a part of validation logic and not ingestion

Comment (tmcgroul) [reply sqa/eth/ingest/ingest.py:226]: as i noticed many libraries that provide validation they just try to substitute missing values

Comment (octo-gone) [reply sqa/eth/ingest/ingest.py:226]: I don't think in this exact case it would be good to move to validation logic. But in general yes!

**Pull request merged**